src/odemis/odemisd/mdupdater.py

```diff
@@ -352,6 +352,11 @@ def observeFilter(self, filter, comp_affected):
         def updateOutWLRange(pos, fl=filter, comp_affected=comp_affected):
             wl_out = fl.axes["band"].choices[pos["band"]]
             comp_affected.updateMetadata({model.MD_OUT_WL: wl_out})
+            # apply lateral chromatic correction to align with the reference channel
+            apply_transform = fl.getMetadata().get(model.MD_CHROMATIC_COR, None)
+            fl_pos_band = fl.position.value["band"]
+            if apply_transform is not None and fl_pos_band in apply_transform.keys():
+                comp_affected.updateMetadata(apply_transform[fl_pos_band])
```

Comment:
As it's coming from a file that can be edited by a user (or service engineer), it'd be good to do some small checks on the content of the MD_CHROMATIC_COR metadata. I'd suggest to check it's a dict, and that `apply_transform[fl_pos_band]` is a dict with all keys of type string. If not, do a logging.error() with complete info about what's the expected format.
Without it, if there is any error in the microscope file, every change of filter will output some quite odd error, looking like it's coming from the filter wheel, potentially hard to diagnose.